src/db/tables/subscribers.js

```diff
@@ -254,11 +277,22 @@ async function deleteUnverifiedSubscribers () {
  * Delete subscriber when a FxA user id is provided
  * Also deletes all the additional email addresses associated with the account
  *
- * @param {string} fxaUID FxA user ID
+ * @param {object} sub subscriber object
  */
-async function deleteSubscriberByFxAUID (fxaUID) {
-  const subscriber = await knex('subscribers').returning('id').where('fxa_uid', fxaUID).del()
-  if (subscriber && subscriber[0]) { await knex('email_addresses').where({ subscriber_id: subscriber[0].id }).del() }
+async function deleteSubscriber (sub) {
+  const trx = await knex.transaction()
+  log.debug('deleteSubscriber', JSON.stringify(sub))
+
+  try {
```

Comment:
Fixing the delete here:
* When we have a secondary that's linked to the primary in `email_addresses` table, we will get a foreign key constraint if we don't delete it first.